src/core/Form/Label/Label.tsx

```diff
@@ -1,34 +1,54 @@
-import React, { Component, ReactNode } from 'react';
+import React, {
+  ReactElement,
+  SetStateAction,
+  Component,
+  ReactNode,
+  useState,
+  isValidElement,
+  cloneElement,
+  CSSProperties,
+} from 'react';
 import classnames from 'classnames';
 import { default as styled } from 'styled-components';
 import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../theme';
 import { baseStyles } from './Label.baseStyles';
 import { asPropType } from '../../../utils/typescript';
 import { VisuallyHidden } from '../../VisuallyHidden/VisuallyHidden';
-import { HtmlSpan, HtmlSpanProps, HtmlDiv, HtmlDivProps } from '../../../reset';
+import {
+  HtmlSpan,
+  HtmlSpanProps,
+  HtmlDivProps,
+  HtmlDivWithRef,
+} from '../../../reset';
 
 export type LabelMode = 'hidden' | 'visible';
 
-export interface LabelProps extends Omit<HtmlDivProps, 'as'> {
-  /** id */
+export interface LabelProps extends Omit<HtmlSpanProps, 'as'> {
+  /** id for label content */
   id?: string;
+  /** Wrapper class name for styling and customizing */
+  className?: string;
   /** Label element content */
   children: ReactNode;
-  /** Custom class name for styling and customizing  */
-  className?: string;
+  /** Content class name for styling and customizing */
+  contentClassName?: string;
+  /** Content style for styling and customizing */
+  style?: CSSProperties;
   /** Hide or show label. Label element is always present, but can be visually hidden.
    * @default visible
    */
   labelMode?: LabelMode;
-  /** Label span props */
-  labelSpanProps?: HtmlSpanProps;
+  /** Props for label wrapper element */
+  wrapperProps?: Omit<HtmlDivProps, 'as' | 'className'>;
   /** Render the wrapping element as another element
    *
    * @default 'label'
    */
   asProp?: asPropType;
   /** Text to mark the field as optional. Shown after labelText and wrapped in parentheses. */
   optionalText?: string;
+  /** Tooltip component for label */
+  tooltipComponent?: ReactElement;
```

Comment:
Instead of `ReactElement` we might be able to be more specific and to support just our own tooltip.

```tsx
tooltipComponent?: typeof Tooltip;
```

Reply:
Shouldn't you support JS as well? If yes, you probably should not remove validity checks and cannot rely on TS types, but rather something that works with JS and React too. Cloning is needed for injecting / merging props so it's probably not possible to remove it.

Reply:
Okay, good notes. Will then leave as it is.